openmc/search.py

```diff
@@ -0,0 +1,249 @@
+from numbers import Real
+from types import FunctionType
+
+import openmc
+import openmc.model
+from openmc.checkvalue import check_type, check_iterable_type, check_length, \
+    check_value
+
+
+_SCALAR_BRACKETED_METHODS = ['brentq', 'brenth', 'ridder', 'bisect']
+
+
+class KeffSearch(object):
+    """Class to perform a keff search by modifying a model parametrized by a
+    single independent variable.
+
+    Parameters
+    ----------
+    model_builder : FunctionType
```

Comment:
It doesn't necessarily *have to* be a function. It could be a class that is callable, so use `collections.Callable` instead